multiversx_sdk_cli/cli_shared.py

```diff
@@ -154,21 +154,11 @@ def check_broadcast_args(args: Any):
 
 
 def check_guardian_args(args: Any):
-    if args.guardian_address:
-        if not args.guardian_service_url or not args.guardian_code:
-            raise errors.BadUsage("All guardian arguments must be provided")
+    if not all([args.guardian, args.guardian_service_url, args.guardian_2fa_code]):
+        raise errors.BadUsage("All guardian arguments must be provided")
 
-    if args.guardian_service_url:
-        if not args.guardian_address or not args.guardian_code:
-            raise errors.BadUsage("All guardian arguments must be provided")
-
-    if args.guardian_code:
-        if not args.guardian_address or not args.guardian_service_url:
-            raise errors.BadUsage("All guardian arguments must be provided")
-
-    if args.guardian_address and args.guardian_service_url and args.guardian_code:
-        if not args.options == 2:
-            raise errors.BadUsage("For guarded transactions the \'options\' must be set to 2")
+    if not args.options >> 1 & 1:
```

Comment:
Works for now, but in the future we should have something like this (below) in `sdk-core`:

https://github.com/multiversx/mx-sdk-js-core/blob/main/src/networkParams.ts#L97

Reply:
Changed